src/csl/mod.rs

```diff
@@ -853,25 +853,39 @@ fn collapse_items<'a, T: EntryLike>(cite: &mut SpeculativeCiteRender<'a, '_, T>)
     match style.citation.collapse {
         Some(Collapse::CitationNumber) => {
             // Option with the start and end of the range.
+            let after_collapse_delim =
+                after_collapse_delim.or(style.citation.layout.delimiter.as_deref());
             let mut range_start: Option<(usize, usize)> = None;
+            let mut ended_range = false;
```

Comment:
I feel like this name needs the word `collapsed` somewhere. Something like `previous_range_collapsed` would be fairly descriptive, but maybe `just_collapsed` is better and less verbose.